### ISSUE TITLE
Require node 10 when install

### ISSUE THREAD

**Issue by anhcao142** [ethereum/web3.js#3548]
#### Expected behavior
Yarn successfully install web3 with node 8
#### Actual behavior
Error when installing web3 with node 8
#### Steps to reproduce the behavior

1. Switch to node 8.11 (using nvm or docker)
2. `yarn add web3`

#### Logs

```
yarn install v1.6.0
[1/4] Resolving packages...
[2/4] Fetching packages...
error mkdirp@1.0.3: The engine "node" is incompatible with this module. Expected version ">=10".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

#### Environment
node from docker node 8.11
web3 version 1.2.8

**Comment by cgewecke** [comment]
@anhcao142 

This issue is discussed in [mkdirp 24][1]. That package has imposed an engine requirement of `>= Node 10` solely because Node 8 has been marked "end of life". It still works though. (We run our [unit tests vs. Node 8][3] and they're fine.)

TLDR; if you're installing with yarn on Node 8 you'll need to use the [--ignore-engines][1] flag

[1]: https://github.com/isaacs/node-mkdirp/issues/24
[2]: https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-ignore-engines
[3]: https://github.com/ethereum/web3.js/pull/3547/checks?check_run_id=716912283

**Comment by cgewecke** [comment]
Closing, please ping if this doesn't work for you for some reason.
